yjit/src/codegen.rs

```diff
@@ -2345,7 +2357,16 @@ fn gen_getlocal_generic(
         // Load the local from the block
         // val = *(vm_get_ep(GET_EP(), level) - idx);
         let offs = -(SIZEOF_VALUE_I32 * ep_offset as i32);
-        Opnd::mem(64, ep_opnd, offs)
+        let local_opnd = Opnd::mem(64, ep_opnd, offs);
+
+        // Write back an argument register to the stack. If the local variable
+        // is an argument, it might have an allocated register, but if this ISEQ
+        // is known to escape EP, the register shouldn't be used after this getlocal.
+        if level == 0 && asm.ctx.get_reg_mapping().get_reg(asm.local_opnd(ep_offset).reg_opnd()).is_some() {
+            asm.mov(local_opnd, asm.local_opnd(ep_offset));
+        }
```

Comment:
I'm a bit confused here.

So we have a situation where some ISEQs escape their environment. We allow getlocal to use a register once and then not again after?

Reply:
> We allow getlocal to use a register once and then not again after?

Yes. We could skip using a register and spill the stack slot early before making the call when YJIT already knows it would escape EP, you would still need this exact same logic in case YJIT doesn't know about it yet, and spilling it early doesn't reduce the number of instructions executed, so I kept the implementation simple by not doing it.